### ISSUE TITLE
problem with urxvt bidi

### ISSUE THREAD

**Issue by AriaMoradi** [mkamensky/Text-Bidi#10]
hi, I'm trying the extension with Persian and there's a bug
when I try to type `ل` and `ا` which should become `لا‍‍`,  things break

**Comment by AriaMoradi** [comment]
![image](https://user-images.githubusercontent.com/8719233/84704541-401b8f00-af6f-11ea-8a54-896d52838172.png)
adding `ا`
![image](https://user-images.githubusercontent.com/8719233/84704614-5b869a00-af6f-11ea-8899-aeadb6e7b993.png)
the text is not shown before added character?

**Comment by AriaMoradi** [comment]
also as visible I have two cursors mirroring  each other across the line
that should be another bug

**Comment by mkamensky** [comment]
> also as visible I have two cursors mirroring each other across the line
> that should be another bug

Yes, I know about this, I don't know how to eliminate this, since there is no way to hide the original cursor (and the author said he would not provide it)

**Comment by mkamensky** [comment]
> hi, I'm trying the extension with Persian and there's a bug
> when I try to type `ل` and `ا` which should become `لا‍‍`, things break

Sorry, I have no experience with this, I think I saw this kind of thing with Hebrew and diactrics, will try to look at it when I have time, thanks for reporting

**Comment by Konfekt** [comment]
@AriaMoradi : Does `ە` and `ا` become `ها` ?

**Comment by Konfekt** [comment]
This problem is due to

> special_encode in rxvtperl.xs

as [discussed](https://github.com/exg/rxvt-unicode/commit/91e9e34407c92474822d06afb5c56419f576a664#diff-bbd4b6a86bc65b6ac8e79e97afc61499158edb40f7bb404c70637b46d80a7ad2) to the  uRxvt developers.

**Comment by mkamensky** [comment]
> This problem is due to
> 
> > special_encode in rxvtperl.xs
> 
> as [discussed](https://github.com/exg/rxvt-unicode/commit/91e9e34407c92474822d06afb5c56419f576a664#diff-bbd4b6a86bc65b6ac8e79e97afc61499158edb40f7bb404c70637b46d80a7ad2) to the uRxvt developers.

Oh, wow, thanks. I guess I'll wait until that is sorted out, before closing here.

**Comment by ghost** [comment]
Hi, Please take a look at the following email and its reponses:
[http://lists.schmorp.de/pipermail/rxvt-unicode/2022q4/002632.html](http://lists.schmorp.de/pipermail/rxvt-unicode/2022q4/002632.html)
I'm not sure if the problem should be solved at the extension level or the uRxvt itself.
